src/core/PrintVisitor.cpp

```diff
@@ -6,29 +6,90 @@
 namespace UML {
 void PrintVisitor::visit(Node *node) { node->accept(this); }
 void PrintVisitor::visit(ClassDecl *classdecl) {
-  std::cout << classdecl->getId() << " details:" << std::endl;
-  std::cout << "------------------------------------" << std::endl;
-  std::cout << "List of attributes" << std::endl;
+  std::string format = R"({ "class" : {
+    "name" : "%s",
+    "attributes" : {
+  )";
+  char formattedStr[2048];
+  sprintf(formattedStr, format.c_str(), classdecl->getId().c_str());
+
+  std::cout << formattedStr;
+
   auto attributes = classdecl->getAttributeList();
-  for (auto attribute : attributes) {
-    attribute->accept(this);
+
+  std::vector<int> variables, methods;
+  int v_delim = 0, m_delim = 0;
```

Comment:
initialize them along with `for` loop. for(int i=0, m_delim=0; i<methods.size(); i++)